src/main/java/frc/robot/subsystems/DriveTrain.java

```diff
@@ -41,4 +44,31 @@ public void arcadeDrive(double move, double rot) {
   public void tankDrive(double left, double right) {
     m_drive.tankDrive(left, right);
   }
+
+  public double getDistance() {
+    // Calculates the average wheel sensor position(for example, if the right wheels did a one full rotation, and the left wheels did a 50% rotation, averageWheels will be 0.75 rotation)
+    double averageWheels = (m_leftFront.getSelectedSensorPosition() + m_leftRear.getSelectedSensorPosition() + m_rightFront.getSelectedSensorPosition() + m_rightRear.getSelectedSensorPosition()) / 4 / Constants.DriveTrainConstants.kMotorToWheelRatio / Constants.DriveTrainConstants.kEncoderResolution; 
+    // the average wheels position * Wheel circumference = overall passed distance
+    return averageWheels * Constants.DriveTrainConstants.kWheelCircumference;
+  }
+
+  public void resetEncoders() {
+    m_leftFront.setSelectedSensorPosition(0);
+    m_leftRear.setSelectedSensorPosition(0);
+    m_rightFront.setSelectedSensorPosition(0);
+    m_rightRear.setSelectedSensorPosition(0);
+  }
+
+  public double getLeftWheelsRPM() {
+    return ((m_leftFront.getSelectedSensorVelocity() * Constants.DriveTrainConstants.k100msTo60sRatio) / 2048 + (m_leftRear.getSelectedSensorVelocity() * Constants.DriveTrainConstants.k100msTo60sRatio / 2048)) / 2 / 4;
```

Comment:
There is a constant for encoder resolution (2048), use it.

Also, the magic **4**, it also has a constant, but do check with the builders that the motor to wheel ratio of the drivetrain's tough box is `4:1`.

~(Pssst... It's `DriveTrainConstants. kEncoderResolution `)~